applications/nrf5340_audio/src/audio/sw_codec_select.c

```diff
@@ -248,26 +255,28 @@ int sw_codec_decode(uint8_t const *const encoded_data, size_t encoded_size, bool
 			break;
 		}
 		case SW_CODEC_STEREO: {
-			if (bad_frame && IS_ENABLED(CONFIG_SW_CODEC_OVERRIDE_PLC)) {
+			if (audio_frame->meta.bad_data &&
+			    IS_ENABLED(CONFIG_SW_CODEC_OVERRIDE_PLC)) {
 				memset(decoded_data_mono[AUDIO_CH_L], 0, PCM_NUM_BYTES_MONO);
 				memset(decoded_data_mono[AUDIO_CH_R], 0, PCM_NUM_BYTES_MONO);
 				decoded_data_size = PCM_NUM_BYTES_MONO;
 			} else {
 				/* Decode left channel */
-				ret = sw_codec_lc3_dec_run(
-					encoded_data, encoded_size / 2, LC3_PCM_NUM_BYTES_MONO,
-					AUDIO_CH_L, decoded_data_mono[AUDIO_CH_L],
-					(uint16_t *)&decoded_data_size, bad_frame);
+				ret = sw_codec_lc3_dec_run(audio_buf->data, (audio_buf->len / 2),
+							   LC3_PCM_NUM_BYTES_MONO, AUDIO_CH_L,
+							   decoded_data_mono[AUDIO_CH_L],
+							   (uint16_t *)&decoded_data_size,
+							   audio_frame->meta.bad_data);
```

Comment:
```suggestion
				frame_size = audio_buf->len / 2 ;
				ret = sw_codec_lc3_dec_run(net_buf_pull_mem(audio_buf->data, frame_size,  frame_size,
							   LC3_PCM_NUM_BYTES_MONO, AUDIO_CH_L,
							   decoded_data_mono[AUDIO_CH_L],
							   (uint16_t *)&decoded_data_size,
							   audio_frame->meta.bad_data);
```

Reply:
By pulling the memory from the net_buf, you advance the data pointer, so that when you want to decode the right channel, `audio_buf->data` will automatically point to the right frame